python/ray/serve/tests/test_http_headers.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+import requests
+from fastapi import FastAPI
+import starlette
+
+from ray import serve
+from ray.serve._private.constants import RAY_SERVE_REQUEST_ID
+
+
+@pytest.mark.parametrize("deploy_type", ["basic", "fastapi", "starlette_resp"])
+def test_http_request_id(ray_shutdown, deploy_type):
+    """Test request id in http headers."""
```

Comment:
please try to be specific in test names & descriptions. in this case I'd go for something like:
```suggestion
def test_user_provided_request_id_header(ray_shutdown, deploy_type):
    """Test that a user-provided request_id is propagated to the replica and returned as a header."""
```

Reply:
please also verify that the passed header is correct inside the replica (using the context)